cwms-radar-client/src/main/java/mil/army/usace/hec/cwms/radar/client/controllers/ProjectEndpointInput.java

```diff
@@ -0,0 +1,175 @@
+        private final String officeId;
+        private final String projectId;
+
+        private GetOne(String projectId, String officeId) {
+            this.projectId = Objects.requireNonNull(projectId, "Cannot retrieve Project data without a project id");
+            this.officeId = Objects.requireNonNull(officeId, "Cannot retrieve Project data without an office id");
+        }
+
+        String projectId() {
+            return projectId;
+        }
+
+        @Override
+        protected HttpRequestBuilder addInputParameters(HttpRequestBuilder httpRequestBuilder) {
+            return httpRequestBuilder.addQueryParameter(OFFICE_QUERY_PARAMETER, officeId)
+                    .addQueryHeader(ACCEPT_QUERY_HEADER, ACCEPT_HEADER_V2);
+        }
+    }
+
+    public static final class GetAll extends EndpointInput {
+        static final String PAGE_SIZE_QUERY_PARAMETER = "page-size";
+        static final String PAGE_QUERY_PARAMETER = "page";
+        private String officeId;
+        private String projectIdMask;
+        private String page;
+        private Integer pageSize;
+
+        private GetAll() {
+            //Empty private ctor
+        }
+
+        @Override
+        protected HttpRequestBuilder addInputParameters(HttpRequestBuilder httpRequestBuilder) {
+            String pageSizeString = Optional.ofNullable(pageSize).map(Object::toString).orElse(null);
+            return httpRequestBuilder.addQueryParameter(OFFICE_QUERY_PARAMETER, officeId)
+                    .addQueryParameter(ID_MASK, projectIdMask)
+                    .addQueryParameter(PAGE_QUERY_PARAMETER, page)
+                    .addQueryParameter(PAGE_SIZE_QUERY_PARAMETER, pageSizeString)
+                    .addQueryHeader(ACCEPT_QUERY_HEADER, ACCEPT_HEADER_V2);
+        }
+
+        public ProjectEndpointInput.GetAll officeId(String officeId) {
+            this.officeId = officeId;
+            return this;
+        }
+
+        public ProjectEndpointInput.GetAll projectIdMask(String levelIdMask) {
+            this.projectIdMask = levelIdMask;
+            return this;
+        }
+
+        public ProjectEndpointInput.GetAll page(String page) {
+            this.page = page;
+            return this;
+        }
+
+        public ProjectEndpointInput.GetAll pageSize(int pageSize) {
+            this.pageSize = pageSize;
+            return this;
+        }
+
+    }
+
+    public static final class Patch extends EndpointInput {
+
+        private final Project project;
+
+        private Patch(Project project) {
+            this.project = Objects.requireNonNull(project, "Cannot patch Project without Project data");
+        }
+
+        Project project() {
+            return project;
+        }
+
+        @Override
+        protected HttpRequestBuilder addInputParameters(HttpRequestBuilder httpRequestBuilder) {
+            return httpRequestBuilder.addQueryParameter(NAME_QUERY_PARAMETER, project.getLocation().getName())
```

Comment:
Why does patch need query parameters for office and location name? shouldn't those be in the Project JSON body?

Reply:
you are right.  removed.